src/pages/index.js

```diff
@@ -20,88 +64,191 @@ const cardSection = new Section(
+      userInfo.setAvatar(updatedData); // Updates avatar 
+      submitButtonTextChanger(avatarPopupSubmit);
+      document.querySelector("#avatar__modal_form").reset(); 
+      avatarPopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(()=>{
+      avatarPopupSubmit.textContent = originalText;
+      console.log('avatar submit button returned to normal')
+    })
+});
+avatarPopup.setEventListeners();
+
+const confirmationPopup = new PopupWithConfirm("#confirmation__modal");
+confirmationPopup.setEventListeners(); 
+
+
+function handleConfirmPopup(card, cardElement) {
+  const confirmPopupOriginal = confirmPopupSubmit.textContent;
+  confirmationPopup.setSubmitFunction(() => {
+    api.deletingCards(card)
+      .then(() => { 
+        submitButtonTextChanger(confirmPopupSubmit);
+        cardElement.remove();
+        confirmationPopup.close();
+      })
+      .catch(err => {console.log("error in handleConfirmPopup:", err)
+      })
+      .finally(()=>{
+        confirmPopupSubmit.textContent = confirmPopupOriginal;
+      })
+  });
+  confirmationPopup.open();
+}
 
-addCardPopup.setEventListeners();
 
 //---------------------CREATING & RENDER CARD--------------------
-
 // Function to create and render a card
 function createCard(cardData) {
-  return new Card(cardData, "#card__template", (data) => imagePopup.open(data)).generateCard();
+  return new Card(
+    cardData, 
+    "#card__template", 
+    (data) => imagePopup.open(data),
+    () => api.likeCard(cardData._id),
+    () => api.dislikeCard(cardData._id),
+    handleConfirmPopup
+    ).generateCard();
 }
 
-// Select DOM elements
-const profileEditBtn = document.getElementById("profile__edit-button");
-const profileNameInput = document.querySelector("[name='title']");
-const profileDescInput = document.querySelector("[name='about']");
-const addModalOpenBtn = document.querySelector(".profile__add-button");
-const pictureNameInput = document.querySelector("[name='name']");
-const pictureUrlInput = document.querySelector("[name='url']")
-const pictureSubmitButton = document.querySelector("#picture_submit");
+//-------------------------Open modal buttons and submit buttons event listenrs---------------------
+
+
 
 profileEditBtn.addEventListener("click", () => {
   const userData = userInfo.getUserInfo();
   console.log("User Info Retrieved:", userData);
-
   profileNameInput.value = userData.title;
   profileDescInput.value = userData.about;
-
   console.log("Name Input Value After Assignment:", profileNameInput.value);
   console.log("Description Input Value After Assignment:", profileDescInput.value);
-
   profileValidator.resetValidation();
   profilePopup.open();
 });
-
 // Event listener for opening the "Add Card" modal
 addModalOpenBtn.addEventListener("click", () => {
+
   addCardValidator.resetValidation(); 
   addCardPopup.open();
 });
 
+avatarEditBtn.addEventListener('click', ()=> {
+  avatarPopup.open();
```

Comment:
* Please, disable the `avatar` submit button   as well https://snipboard.io/9b8Ulp.jpg 


```js

avatarEditBtn.addEventListener("click", () => {
  avatarValidator.resetValidation();
  avatarPopup.open();
});

```